docs/extend/extend-cli/cli-implement-profiles.md

```diff
@@ -1,74 +1,106 @@
+            properties: {
+              host: {
+                type: "string",
+                optionDefinition: {
+                  type: "string",
+                  name: "host",
+                  alias:["H"],
+                  description: "Host name of your SOME PRODUCT REST API server"
+                }
+              },
+              port: {
+                type: "number",
+                optionDefinition: {
+                  type: "number",
+                  name: "port",
+                  alias:["P"],         
+                  includeInTemplate: true,
+                  description: "Port number of your SOME PRODUCT REST API server"
+                }
+              },
+              user: {
+                type: "string",
+                optionDefinition: {
+                  type: "string",
+                  name: "user",
+                  alias:["u"],          
+                  description: "User name to authenticate to your SOME PRODUCT REST API server"
+                },
+                secure: true
+              },
+              password: {
+                type: "string",
+                optionDefinition: {
+                  type: "string",
+                  name: "password",
+                  alias:["p"],          
+                  description: "Password to authenticate to your SOME PRODUCT REST API server"
+                },
+                secure: true
+              },
             },
-            secure: true
           },
-        },
-      },
-      createProfileExamples: [
-        {
-         options: "spprofile --host zos123 --port 1234 --user ibmuser --password myp4ss",
-         description: "Create a SOME PRODUCT profile named 'spprofile' to connect to SOME PRODUCT at host zos123 and port 1234"
         }
-      ]
-    }
-]
-```
+    ]
+    ```
+3. Replace `someproduct` in `type: "someproduct"` with your plug-in profile name.
+
+    This property represents the name of the profile type option that might be required on commands that allow users to select a profile from their team configuration. (Zowe CLI has profile types for z/OSMF, SSH, and TSO.)
+
+      - For example, the `zowe zosmf check status --zosmf-profile` command includes an option that specifies a z/OSMF profile from the user's team configuration. For your plug-in, the profile type that you define can be used in a commands to specify a profile for your plug-in.
+    
+    This value is used in team configuration to create a plug-in profile to connect to the mainframe.
+
+4. Update the title and description for your plug-in.
+
+5. Add any needed properties to allow users to configure plug-in settings.
+
+    Each option property in the template includes an `optionDefinition`. For a reference of `optionDefinition` attributes, see the [`ICommandOptionDefinition` interface](https://github.com/zowe/zowe-cli/blob/master/packages/imperative/src/cmd/src/doc/option/ICommandOptionDefinition.ts).
```

Comment:
For the `ICommandOptionDefinition` URL, should we link to our typedoc? https://docs.zowe.org/stable/typedoc/interfaces/_zowe_imperative.icommandoptiondefinition.html

Not sure which we prefer - pointing directly to the source code or to docs that are auto-generated from it.

Reply:
Leaving it as is.